rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/jvminterop/JavaSystemInterface.kt

```diff
@@ -50,8 +52,23 @@ open class JavaSystemInterface(
     }
 
     // For calls from Java programs
-    private constructor (os: PrintStream, rpgSystem: RpgSystem) : this(os, rpgSystem::getProgram, { copyId -> rpgSystem.getCopy(copyId) }, rpgSystem = rpgSystem)
-    constructor (os: PrintStream) : this(os, RpgSystem())
+    private constructor (os: PrintStream, rpgSystem: RpgSystem, configuration: Configuration?) : this(os, rpgSystem::getProgram, { copyId -> rpgSystem.getCopy(copyId) }, rpgSystem = rpgSystem, configuration = configuration)
+    /**
+     * Creates an instance of JavaSystemInterface with default [RpgSystem]
+     * */
+    constructor (os: PrintStream, configuration: Configuration?) : this(os, RpgSystem(), configuration)
+    /**
+     * Creates an instance of JavaSystemInterface with default [RpgSystem]
+     * */
+    constructor (os: PrintStream) : this(os, RpgSystem(), null)
+    /**
+     * Creates an instance of JavaSystemInterface with default [RpgSystem] and os param set to [System.out]
+     * */
+    constructor(configuration: Configuration) : this(System.out, configuration)
+
+    /**
+     * Creates an instance of JavaSystemInterface with default [RpgSystem] and os param set to [System.out]
```

Comment:
Again, I don't understand from this comment the difference in logic from the previous statement. Could you explain it to me?

Reply:
In this case you can create an instance of JavaSystemInterface by passing only the parameter configuration.